PSReadLine/DynamicHelp.cs

```diff
@@ -0,0 +1,288 @@
+                AdjustForPossibleScroll(1);
+                MoveCursorDown(1);
+
+                var bufferWidth = console.BufferWidth;
+                var columnWidth = this.ColumnWidth;
+
+                var items = this.HelpItems;
+                for (var row = 0; row < this.Rows; row++)
+                {
+                    var cells = 0;
+                    for (var col = 0; col < this.Columns; col++)
+                    {
+                        var index = row + (this.Rows * col);
+                        if (index >= items.Count)
+                        {
+                            break;
+                        }
+                        console.Write(GetItem(items[index], columnWidth));
+                        cells += columnWidth;
+                    }
+
+                    // Make sure we always write out exactly 1 buffer width to erase anything
+                    // from a previous menu.
+                    if (cells < bufferWidth)
+                    {
+                        // 'BlankRestOfLine' erases rest of the current line, but the cursor is not moved.
+                        console.BlankRestOfLine();
+                    }
+
+                    // Explicit newline so consoles see each row as distinct lines, but skip the
+                    // last line so we don't scroll.
+                    if (row != (this.Rows - 1) || !menuSelect)
+                    {
+                        AdjustForPossibleScroll(1);
+                        MoveCursorDown(1);
+                    }
+                }
+
+                bool extraPreRowsCleared = false;
+                if (previewBlock != null)
+                {
+                    if (Rows < previewBlock.Rows)
+                    {
+                        // If the last menu row took the whole buffer width, then the cursor could be pushed to the
+                        // beginning of the next line in the legacy console host (NOT in modern terminals such as
+                        // Windows Terminal, VSCode Terminal, or virtual-terminal-enabled console host). In such a
+                        // case, there is no need to move the cursor to the next line.
+                        //
+                        // If that is not the case, namely 'CursorLeft != 0', then the rest of the last menu row was
+                        // erased, but the cursor was not moved to the next line, so we will move the cursor.
+                        if (console.CursorLeft != 0)
+                        {
+                            // There are lines from the previous rendering that need to be cleared,
+                            // so we are sure there is no need to scroll.
+                            MoveCursorDown(1);
+                        }
+
+                        Singleton.WriteBlankLines(previewBlock.Rows - Rows);
+                        extraPreRowsCleared = true;
+                    }
+                }
```

Comment:
Here too. The `previewBlock` thing is a copy from `Menu`. It exists in `Menu` because `Menu` is interactive and can change when selection is changed. For `MutilineBlockDisplay`, it's not interactive and won't change after rendering, so the `previousBlock` doesn't make sense at all.

Reply:
refactored